src/tactic/ring.lean

```diff
@@ -386,7 +386,7 @@ meta def eval : expr → ring_m (horner_expr × expr)
     (e', p) ← lift $ norm_num.derive e,
     lift $ e'.to_rat,
     return (const e', p)) <|> eval_atom e
-| `(%%e₁ / %%e₂) := do
+| `(@has_div.div _ division_ring_has_div %%e₁ %%e₂) := do
```

Comment:
Do you think the syntactic check here is too strong? What about matching the instance as `%%inst` and checking that it unifies with an application of `division_ring_has_div`?

Reply:
@robertylewis I suppose that's a possibility, but I don't actually think it comes up; there aren't any other competing instances for `has_div` that I'm aware of, everything filters through this instance. (There are many ways to prove the resulting `division_ring` goal, but AFAIK the only `has_div` instances are the ones for nat/int and the division ring div.)

Reply:
I dunno, that doesn't sound very robust. Is the current Travis failure related? (I'm not gonna compile it on my laptop right now.)

Reply:
There's another instance called `division_ring_has_div'`. This is causing the current travis failure.